### PR TITLE
Added healthcheck support in container inspect JSON result

### PR DESCRIPTION
**Pull request by shobull**
Docker version 1.12 introduced [healthcheck support](https://docs.docker.com/engine/reference/builder/#/healthcheck). Healthcheck status is reported in docker inspech result. Result is there only if health check was used.

This PR includes unit test for parsing of JSON docker inspect report.

---
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/763)

### CODE REVIEW

File: src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java [hunk 1]
```diff
@@ -390,158 +559,25 @@ public String getFinishedAt() {
...
         }
 
-        @Override
-        public String toString() {
-            return ToStringBuilder.reflectionToString(this);
-        }
-    }
-
-    @JsonIgnoreProperties(ignoreUnknown = true)
-    public static class Mount {
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java]: Why do you reorder classes?

Comment (shobull) [inline src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java]: Sorry for that, my IDE did autoformatting and I didn't noticed.

File: src/test/java/com/github/dockerjava/api/command/CommandJSONSamples.java [hunk 2]
```diff
@@ -24,7 +24,7 @@
  */
 public enum CommandJSONSamples implements JSONResourceRef {
 
-    inspectContainerResponse_full, inspectContainerResponse_full_1_21, inspectContainerResponse_empty;
+    inspectContainerResponse_full, inspectContainerResponse_full_1_21, inspectContainerResponse_empty, inspectContainerResponse_full_healthcheck;
```
Comment (KostyaSha) [inline src/test/java/com/github/dockerjava/api/command/CommandJSONSamples.java]: old deprecated way of testing. Please use test files directly via helper methods as it done in https://github.com/docker-java/docker-java/blob/419900d074e6d9754dcd5c6290d93c3350bab010/src/test/java/com/github/dockerjava/api/model/EventsTest.java#L25

File: src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java [hunk 3]
```diff
@@ -215,12 +211,178 @@ public String getMountLabel() {
...
+    public static class HealthState {
+
+        @JsonProperty("Status")
+        private String status;
+
+        @JsonProperty("FailingStreak")
+        private Integer failingStreak;
+
+        @JsonProperty("Log")
+        private String[] log;
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java]: No arrays, use List

**Comment by cdancy** [comment]
Any updates here? @KostyaSha anything I can do to move this forward? Actually have a need for this now.

**Comment by cdancy** [comment]
@KostyaSha @shobull any thoughts on my comment above? Any way I can help move this forward?

**Comment by KostyaSha** [comment]
@cdancy i usually pick PR head, rebase on upstream/master, then add my changes and run `git rebase -i HEAD~5` then place fixup. That merges contributors and keep them as contributors.
```
[remote "upstream-pr"]
	url = https://github.com/docker-java/docker-java.git
	fetch = +refs/pull/*:refs/remotes/upstream-pr/pull/*
```
Pick pr and open new PR linking to this pr with word `closes #763` when you pass review i will merge. If there is something needed in master, then i can kick release.

**Comment by KostyaSha** [comment]
https://github.com/docker-java/docker-java/blob/master/docs/devel.adoc

**Comment by cdancy** [comment]
@KostyaSha oh yeah I got all of that just wondering what's left to be done. From what I can tell it's just changing the use of array to list above. Anything else missing?

**Comment by KostyaSha** [comment]
Maybe move nested class outside.  may be call it`ContainerHealthCheck`

File: src/test/java/com/github/dockerjava/api/command/CommandJSONSamples.java (hunk 2 above)
Comment (KostyaSha) [inline src/test/java/com/github/dockerjava/api/command/CommandJSONSamples.java]: @cdancy also this

File: src/test/resources/com/github/dockerjava/api/command/inspectContainerResponse_full_healthcheck.json [hunk 4]
```diff
@@ -0,0 +1,156 @@
+[
```
Comment (KostyaSha) [inline src/test/resources/com/github/dockerjava/api/command/inspectContainerResponse_full_healthcheck.json]: i made new apis for json dumps, there is clear structure with versions and places from what they come. So this also need to be moved.

**Comment by cdancy** [comment]
> Maybe move nested class outside. may be call itContainerHealthCheck

@KostyaSha think dev was following standard set by other inner classes like `Mount` and `ContainerState`. Seems to make sense to keep it within `InspectContainerResponse` no?

**Comment by cdancy** [comment]
> @KostyaSha think dev was following standard set by other inner classes like Mount and ContainerState. Seems to make sense to keep it within InspectContainerResponse no?

Actually I take this back. Seems the response is more complicated than the dev had realized. Splitting into its own file makes sense. Stay tuned

**Comment by KostyaSha** [comment]
Fixed via https://github.com/docker-java/docker-java/pull/805

**Pull request closed without merging**